### ISSUE TITLE
All in One SEO 

### ISSUE THREAD

**Issue by markus-mr** [qtranslate/qtranslate-xt#933]
Dear qtranslate-xt Team,

In the new version of "All in One SEO" its not possible to switch between the languages anymore.
I use "qtranslate-xt" with [:de]XXXXX [:en]XXXXX[:]

Snippet Preview before read correctly the [:de]XXXXX [:en]XXXXX[:] codes to normal texts and it was possible to switch between the languages. Now, the Snippet Preview shows only [:de]XXXXX [:en]XXXXX[:] and its not more possible to switch between the languages.
I hope Google can still work work this. Correction of 800pages will be extreme. 

Please help.
Best!

**Comment by herrvigg** [comment]
Did it work with version 4.0.10?

**Comment by herrvigg** [comment]
Or better formulated, what is the last version for which it was still working?

**Comment by markus-mr** [comment]
> Did it work with version 4.0.10?

Yes, the last version was great, was still working. Snippet Preview and Switch of languages is not more possible.
I hope google will don't change it in the next days to [:de]XXXXX[:en]XXXXX[:]. Hopefully it continues to display it correctly.

Best!

**Comment by herrvigg** [comment]
Are you sure it was really 4.0.10? This would be really surprising.
Imo, everything is obsolete since version 4, see #930.
Every item that had id `aiosp_` does not exist anymore, for example `aiosp_title`. But `aioseop_snippet_title` is also gone.

if you add this in qtranslate-xt/modules/all-in-one-seo-pack/qaioseop-admin.php, line 23:
```
    $fields[] = array( 'jquery' => '.site-title', 'encode' => 'display' );
```
You should see the title preview translated on page reload. But it won't be updated if the code tags are changed.

In any case, this is for display only in the admin section. This is not grabbed by Google SEO for example. Here is the real question, how should this be integrated with qTranslate? Maybe there is a specific hook that should be called.

**Comment by bagaweb** [comment]
> 
>
> Did it work with version 4.0.10?

Last AIOSP working is 3.7.1

> 
> 
> if you add this in qtranslate-xt/modules/all-in-one-seo-pack/qaioseop-admin.php, line 23:
> 
> ```
>     $fields[] = array( 'jquery' => '.site-title', 'encode' => 'display' );
> ```


following your example I also added this line for meta description:
 ```
     $fields[] = array( 'jquery' => '.meta-description', 'encode' => 'display' );
 ```

**Comment by markus-mr** [comment]
My last good working AIOSP Version was with the old layout. Don't know which version it was.

Thanks for all help. I try the php codes later.

Best!

**Comment by valerensi** [comment]
Hello, 
I jump in this topic to tell you that the problem is still present. 
In this moment with the latest version of qtranslate-xt (v. 3.11.0) and All in One SEO (v. 4.1.3.3) the language switch in Wordpress backend not working so we are forced to insert the meta description using [:it][:en][:] tag.
This occurs both with All in One SEO free or pro version. 
This should be a problem for our clients :(
Is there any solution? If you want we can send you some screenshot or video if it should helps!
Let me know! Thanks.

**Comment by herrvigg** [comment]
I can't really answer for the specific question about All in One SEO, but more generally some parts are very difficult to maintain and I can understand the pain for the end users.
qTranslate-XT has been designed as a "hack". It's adapted for small websites (which I use it for) but less for customers having high expectations. Still, it can do the job for many and it can evolve. That's where the developers community has a role to play.

We don't have any good solution for dynamic fields yet, because everything must be handled by events and the third-party plugins usually don't provide anything. 

In fact it feels we'd need a "tag editor" that could work as a generic workaround for many issues like that.
The idea would be to enter the languages separately and then copy-paste the raw values generated.
It would not solve it completely but it could make it a bit easier while waiting for a better solution that could take very long....

**Comment by valerensi** [comment]
I know you are working hard to keep the plugin up and running. Your proposal for workaround would certainly be a good idea. Unfortunately the other language plugins are less effective and usable than this one so we would be sorry to abandon it! Thanks for your answer.

**Comment by jurgen-74** [comment]
> Hello, I jump in this topic to tell you that the problem is still present. In this moment with the latest version of qtranslate-xt (v. 3.11.0) and All in One SEO (v. 4.1.3.3) the language switch in Wordpress backend not working so we are forced to insert the meta description using [:it][:en][:] tag. This occurs both with All in One SEO free or pro version. This should be a problem for our clients :( Is there any solution? If you want we can send you some screenshot or video if it should helps! Let me know! Thanks.

So, do I understand you correctly that you actually use the tags [:en] and [:] in the separate AIOSEO fields?
If that is the case, doesn't that mess up the other fields (in the page or post)?

Also, I didn't update AIOSEO because of this reason (I still work with version 3.7.1). But, inserting tags isn't a big deal for me. If I update AIOSEO to the newest version, will the meta of the old version still function? I mean, will I still have two separate languages in my old posts and page's SEO?

Sorry for asking, but I do not have a test platform, and I do not want to try something like this before I know it will work.

Blessings,
Jurgen

**Comment by valerensi** [comment]
Hello @jurgen-74 at this moment there are two options:

- use the predefined tags for AIOSEO "eg title, excerpt, content" => this allows you to fill in the values correctly but with minimal customization
- use custom texts for AIOSEO => to have them correctly interpreted then in the source code it is necessary to use these parameters [: en] [: it] [:]

This now means that if my exerpt is too long or not suitable for SEO because it does not contain keywords or terms useful for positioning, I have to personalize that value with tags [: en] [: it] etc. The method works but it is not very easy for the customer to do it. They do not create problems with the other fields on the page because they are completely ignored, AIOSEO fields are treated as a single field common to all languages and this is precisely the problem. In previous versions this did not happen. The AIOSEO title and description fields were split and switched in backend exactly as it happens for all the other WP editing fields for every language added by qtranslate.

**Comment by jurgen-74** [comment]
Thank you @valerensi for answering my question. My apologise for not big as quick (our internet is often very bad... Can't even open email). So, if I update to the newest version of AIOSEO, I could continue using two languages.

> use the predefined tags for AIOSEO "eg title, excerpt, content" => this allows you to fill in the values correctly but with minimal customization

I don't think I understand this option. Do you mean to use these tags in the text editor of wordpress itself or in a AIOSEO field?

> use custom texts for AIOSEO => to have them correctly interpreted then in the source code it is necessary to use these parameters [: en] [: it] [:]

Following this option, I have to manually put it in qtranslate-xt parameters: `[:nl]Wat hebben we een lol[:en]What fun we have![:] `
In other words, the new version of AIOSEO will not have separate fields for different languages, but it will just have one field. And in that _one_ field I can write my excerpt with the parameters `[:nl] [:en] [:]`

The second option seems to be most easy one for me personally. So, one last question. Will the pages on which I used the old version of AIOSEO (with the two languages) change when I update to the newest version?

Thanks again,
Jurgen

**Comment by valerensi** [comment]
Hello,
I think is simpler to show than to explain.

Case 1:
[https://www.dropbox.com/s/r34mep0pdg8nmgl/AIOSEO-tag.png?dl=0]
this is the way to populate content for SEO with Wordpress fields.
the output correctly works with all languages

Case 2: 
[https://www.dropbox.com/s/0km2ugvmsx7id17/AIOSEO-compiled.png?dl=0]
this is the only solution to customize contents for different languages without using Wordpress fields.
the output correctly works with all languages, but it's difficult for a client to use and understand.
Also it has to be considering that the count of words is no longer useful in this way.
